dali/test/python/test_operator_readers_numpy.py

```diff
@@ -356,3 +357,142 @@ def test_numpy_reader_alias():
+
+
+def _testimpl_numpy_reader_roi_error(file_root, batch_size, ndim, dtype, device, fortran_order=False, file_filter="*.npy",
+                                     roi_start=None, rel_roi_start=None, roi_end=None, rel_roi_end=None, roi_shape=None,
+                                     rel_roi_shape=None, roi_axes=None, out_of_bounds_policy=None, fill_value=None):
+    @pipeline_def(batch_size=3, device_id=0, num_threads=8)
+    def pipe():
+        print(file_root)
+        print(file_filter)
+        data = fn.readers.numpy(device=device, file_root=file_root, file_filter=file_filter,
+                                roi_start=roi_start, rel_roi_start=rel_roi_start,
+                                roi_end=roi_end, rel_roi_end=rel_roi_end,
+                                roi_shape=roi_shape, rel_roi_shape=rel_roi_shape,
+                                roi_axes=roi_axes, out_of_bounds_policy=out_of_bounds_policy,
+                                fill_value=fill_value,
+                                shard_id=0, num_shards=1, cache_header_information=False)
+        return data
+    p = pipe()
+    err = None
+    try:
+        p.build()
+        p.run()
+    except RuntimeError as thrown:
+        err = thrown
+    # asserts should not be in except block to avoid printing nested exception on failure
+    assert err, "Exception not thrown"
+
+def test_numpy_reader_roi_error():
+    # setup file
+    shapes=[(10, 10), (12, 10), (10, 12), (20, 15), (10, 11), (12, 11), (13, 11), (19, 10)]
+    ndim=2
+    dtype=np.uint8
+    batch_size=8
+    file_filter="*.npy"
+
+    # roi_start, rel_roi_start, roi_end, rel_roi_end, roi_shape, rel_roi_shape, roi_axes, out_of_bounds_policy
+    roi_args = [
+        ([1, 2], [0.1, 0.2], None, None, None, None, None, None),  # Both roi_start and rel_roi_start
+        (None, None, [8, 7], [0.4, 0.5], None, None, None, None),  # Both roi_end and rel_roi_end
+        (None, None, [8, 7], None, [8, 7], None, None, None),  # Both roi_end and roi_shape
+        (None, None, [8, 7], None, None, [0.4, 0.5], None, None),  # Both roi_end and rel_roi_shape
+        (None, None, None, [0.5, 0.4], [8, 7], None, None, None),  # Both rel_roi_end and roi_shape
+        ([-1, 2], None, None, None, None, None, None, None), # Out of bounds anchor
+        (None, None, [100, 8], None, None, None, None, None), # Out of bounds end
+        (None, None, None, None, [100, 8], None, None, None), # Out of bounds shape
+    ]
+    fortran_order = False
+    with tempfile.TemporaryDirectory(prefix = gds_data_root) as test_data_root:
+        index = 0
+        for sh in shapes:
+            filename = os.path.join(test_data_root, "test_{:02d}.npy".format(index))
+            create_numpy_file(filename, sh, dtype, fortran_order=fortran_order)
+
+        for device in ["cpu"]:  # TODO(janton): ROI is only supported for CPU backend, for now
```

Comment:
Could you add a negative case for GPU as well?

Reply:
Will do